tests/test_client.py

```diff
@@ -49,3 +52,37 @@ def test_invalid_credentials(
 
         with pytest.raises(error_name):
             client.delete_documents(consent_id=consent_id)
+
+
+@pytest.mark.parametrize('content', [
+    __file__,
+    Path(__file__),
+    open(__file__, 'rb'),
+    open(__file__, 'r'),
+    open(__file__),
+    Path(__file__).open(),
+    Path(__file__).open('rb'),
+    Path(__file__).open('r'),
+    Path(__file__).read_bytes(),
+    Path(__file__).read_text().encode(),
+    b64encode(Path(__file__).read_bytes()),
+    bytearray(Path(__file__).read_bytes()),
+])
+def test_parse_content(content):
+    expected_result = b64encode(Path(__file__).read_bytes()).decode()
+    result = parse_content(content)
+    assert result == expected_result
+
+
+@pytest.mark.parametrize(('content', 'error'), [
+    (uuid.uuid4().hex, FileNotFoundError),
```

Comment:
I suggest using a new exception type `ParseError` or similar which will be raised when `parse_content` fails.

Reply:
Why? Isn't it better with a more specific error depending on what you input? 
As a user it seems more useful to get a FileNotFoundError which tells you very clearly that the file you where trying to parse doesn't exist rather than a ParseError which could mean anything.

Reply:
I also agree with @magnurud that it seems more helpful to get the actual error thrown instead. Why do you believe creating a ParseError is better in this case @krigar1184 ?